### PR TITLE
Update stylings of error states

### LINKED ISSUE #7434: [North Star Applicant UI] Add error handling styles
This issue covers adding error states for all question types in the applicant flow. We need to handle:

1) Surfacing error messages
2) Error styles for all single text input
3) Error styles for more complicated inputs (multi input questions etc)

This doesn't cover the work to add more precise validation for date questions

### PR DESCRIPTION
**Pull request by greg-hyde**
### Description

Adds error styling for question types other than single text inputs. Adds an error bar per input. There are some small padding changes that impact the questions that were done in a previous PR but those stayed mostly the same

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### User visible changes

- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order

#### New Features

- [x] Conditioned new functionality on a [FeatureFlag](https://github.com/civiform/civiform/wiki/Feature-Flags)
- [x] Wrote browser tests with the feature flag off and on, etc.

### Issue(s) this completes

Fixes #7434

### CODE REVIEW

**Comment by greg-hyde** [comment]
* **#7420** <a href="https://app.graphite.dev/github/pr/civiform/civiform/7420?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @greg-hyde and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: server/app/assets/stylesheets/uswds/_uswds-theme-custom-styles.scss [hunk 1]
```diff
@@ -39,10 +39,18 @@ See the radio-option below as an example.
...
+.cf-question-header {
   @include u-padding-left(2)
 }
+
+/*
+ * Less padding when there is an error, since we add a border and still want the inputs to line up.
+ */
+ .cf-applicant-question-field.cf-question-field-with-error {
+  @include u-padding-left('105')
+}
```
Comment (avaleske) [inline server/app/assets/stylesheets/uswds/_uswds-theme-custom-styles.scss:56]: *nit*: newline at the end of the file pls

**Review by avaleske — APPROVED** [review]
What is our guideline about when to use USWDS utilities vs when to use tailwind utilties?

**Comment by greg-hyde** [comment]
> What is our guideline about when to use USWDS utilities vs when to use tailwind utilties?

I've been trying to use USWDS styles for Thymeleaf since we're using USWDS components and it provides some consistency there and a path towards theming, but for most classes we could honestly use tailwind or uswds or vanilla CSS and it wouldn't really make a difference

**Review by avaleske — APPROVED** [review]

**Pull request merged**